### DISCUSSION TITLE
Why clamp in DenseSageConv?

### DISCUSSION THREAD

**Discussion by mdanb** [pyg-team/pytorch_geometric#2507] [Q&A] [answered]
In the `forward` method for `DenseSAGEConv`, there is the line:
```out = out / adj.sum(dim=-1, keepdim=True).clamp(min=1)```
I understand that clamping will prevent division by 0 if the degree of a node is 0, but what if the edges are weighted, in which case the "degree" can be between 0 and 1?

**Comment by rusty1s** [comment, marked as answer]
Adjacency values in `DenseSAGEConv` should not be weighted, and one should use `DenseGraphConv` instead. I need to add this to the documentation.
